File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.cpp
class Solution {
public:
  int concatenatedBinary(int n) {
    long ret = 0;
    int shift = 0;
    int MOD = 1e9+7;
    for(int i=1; i<=n; i++){
      if((i&(i-1))==0)
        shift++;
      ret = ((ret<<shift)|i)%MOD;
    }
    return ret;
  }
};

File: 1680-concatenation-of-consecutive-binary-numbers/1680-concatenation-of-consecutive-binary-numbers.py
class Solution:
    def concatenatedBinary(self, n: int) -> int:
      ret = 0
      shift = 0
      MOD = int(1e9+7)
      for i in range(1, n+1):
        if(i&(i-1) == 0):
          shift+=1
        ret = ((ret<<shift)|i)%MOD
      return ret